src/Npgsql/PostgresTypes/PostgresBaseType.cs

```diff
@@ -35,11 +37,100 @@ namespace Npgsql.PostgresTypes
     /// </summary>
     public class PostgresBaseType : PostgresType
     {
-        /// <summary>
-        /// Constructs an unsupported base type (no handler exists in Npgsql for this type)
-        /// </summary>
-        protected internal PostgresBaseType(string ns, string name, uint oid)
-            : base(ns, name, oid)
+        /// <inheritdoc/>
+        protected internal PostgresBaseType(string ns, string internalName, uint oid)
+            : base(ns, TranslateInternalName(internalName), internalName, oid)
         {}
+
+        /// <inheritdoc/>
+        internal override string GetPartialNameWithFacets(int typeModifier)
+        {
+            var facets = GetFacets(typeModifier);
+            if (facets == PostgresFacets.None)
+                return Name;
+
+            switch (Name)
+            {
+            // Special case for timestamptz and timetz where the facet is embedded in the middle
+            case "timestamp with time zone":
+                return $"timestamp{facets} with time zone";
+            case "time with time zone":
+                return $"time{facets} with time zone";
+            default:
+                return $"{Name}{facets}";
+            }
+        }
+
+        [CanBeNull]
+        internal override PostgresFacets GetFacets(int typeModifier)
```

Comment:
`PostgresFacets` is a struct. It can't be `null`.

Reply:
Yep, was a leftover from an earlier version.